ports/stm/mpconfigport.h

```diff
@@ -44,27 +44,6 @@ extern uint8_t _ld_default_stack_size;
 
 #include "py/circuitpy_mpconfig.h"
 
-// The STM32 HAL file is included virtually everywhere:
-#include STM32_HAL_H
```

Comment:
Why take this out? Makes it have to be manually included with every clocks.c file. And it doesn't impact the common-hal modules in your testing?

Reply:
Manually including it is a better practice because it's clearer where definitions are coming from then. Eventually I'd like to use include-what-you-use to automate include maintenance: https://include-what-you-use.org/

Everything compiled ok so I suspect common-hal code is getting it implicitly.